apps/prairielearn/assets/scripts/lib/histogram.ts

```diff
@@ -0,0 +1,141 @@
+    ylabel?: string;
+    xTickLabels?: string[] | 'auto';
+    topMargin?: number;
+    rightMargin?: number;
+    bottomMargin?: number;
+    leftMargin?: number;
+  },
+) {
+  options = {
+    width: 100,
+    height: 40,
+    xmin: 'auto',
+    xmax: 'auto',
+    ymin: 'auto',
+    ymax: 'auto',
+    xlabel: 'value',
+    ylabel: 'count',
+    xTickLabels: 'auto',
+    topMargin: 10,
+    rightMargin: 20,
+    bottomMargin: 55,
+    leftMargin: 70,
+    ...options,
+  };
+
+  const width = 600 - (options.leftMargin ?? 70) - (options.rightMargin ?? 20);
+  const height = 371 - (options.topMargin ?? 10) - (options.bottomMargin ?? 55);
+
+  const xmin = options.xmin === 'auto' ? Math.min(...xgrid) : options.xmin;
+  const xmax = options.xmax === 'auto' ? Math.max(...xgrid) : options.xmax;
+  const x = window.d3.scaleLinear().domain([xmin, xmax]).range([0, width]);
+
+  const ymin = options.ymin === 'auto' ? Math.min(...data) : options.ymin;
+  const ymax = options.ymax === 'auto' ? Math.max(...data) : options.ymax;
+  const y = window.d3.scaleLinear().domain([ymin, ymax]).nice().range([height, 0]);
+
+  const xTickFormat =
+    options.xTickLabels === 'auto'
+      ? null
+      : function (d: null, i: number) {
+          return options.xTickLabels ? options.xTickLabels[i] : null;
+        };
+  const xAxis = window.d3.axisBottom().scale(x).tickValues(xgrid).tickFormat(xTickFormat);
+
+  const yAxis = window.d3.axisLeft().scale(y);
+
+  const xGrid = window.d3.axisBottom().scale(x).tickValues(xgrid).tickSize(-height).tickFormat('');
+
+  const yGrid = window.d3.axisLeft().scale(y).tickSize(-width).tickFormat('');
+
+  const svg = window.d3
+    .select($(selector).get(0))
+    .append('svg')
+    .attr('width', width + (options.leftMargin ?? 0) + (options.rightMargin ?? 0))
+    .attr('height', height + (options.topMargin ?? 0) + (options.bottomMargin ?? 0))
+    .attr('class', 'center-block statsPlot')
+    .append('g')
+    .attr('transform', 'translate(' + options.leftMargin + ',' + options.topMargin + ')');
+
+  svg
+    .append('g')
+    .attr('class', 'x grid')
+    .attr('transform', 'translate(0,' + height + ')')
+    .call(xGrid);
+
+  svg.append('g').attr('class', 'y grid').call(yGrid);
+
+  svg
+    .selectAll('.outlineBar')
+    .data(data) // .data(_.times(data.length, _.constant(0)))
+    .enter()
+    .append('rect')
+    .attr('class', 'outlineBar')
+    .attr('x', function (d: null, i: number) {
+      return x(xgrid[i]);
+    })
+    .attr('y', function (d: number) {
+      return y(d);
+    })
+    .attr('width', function (d: null, i: number) {
+      return x(xgrid[i + 1]) - x(xgrid[i]);
+    })
+    .attr('height', function (d: number) {
+      return y(0) - y(d);
+    });
+
+  /*
+        svg.selectAll(".outlineBar")
+        .data(data)
+        .transition()
+        .duration(3000)
+        .attr("y", function(d, i) {return y(d);})
+        .attr("height", function(d, i) {return y(0) - y(d);});
+      */
```

Comment:
Is this needed?

Reply:
No, removed in [2294ce6](https://github.com/PrairieLearn/PrairieLearn/pull/9809/commits/2294ce6fe8294dda929e0be0563c3e2316102eac).